source/CLEO_SDK/demo_plugins/IniFiles/dllmain.cpp

```diff
@@ -181,29 +181,43 @@ Opcode Format
 0AF4=4,%4d% = read_string_from_ini_file %1s% section %2s% key %3s%
 ****************************************************************/
 {
-	script->Collect(3);
-
 	char iniPath[MAX_PATH];
 	char path[100];
 	char sectionName[100];
 	char key[100];
 	char strValue[100];
-	BOOL result;
 
-	strcpy(path, Params[0].cVar);
+	script->Collect(3);
+  	strcpy(path, Params[0].cVar);
 	strcpy(sectionName, Params[1].cVar);
 	strcpy(key, Params[2].cVar);
 
 	//if path is short, GetPrivateProfileInt() searches for the file in the Windows directory
 	MakeFullPath(path, iniPath);
 
-	result = GetPrivateProfileString(sectionName, key, NULL, strValue, sizeof(strValue), iniPath);
-
-	strcpy(Params[0].cVar, strValue);
-	script->Store(1);
+	DWORD readCount = GetPrivateProfileString(sectionName, key, NULL, strValue, sizeof(strValue), iniPath);
+	if (readCount == 0)
+	{
+		script->Collect(1); // skip result param
+		script->UpdateCompareFlag(false);
+		return OR_CONTINUE;
+	}
 
-	script->UpdateCompareFlag(result != 0);
+	auto resultType = script->GetNextParamType();
+	switch (resultType)
+	{
+	// pointer to target buffer
+	case eParamType::PARAM_TYPE_LVAR:
+	case eParamType::PARAM_TYPE_GVAR:
+		script->Collect(1); 
+		strcpy(Params[0].cVar, strValue);
```

Comment:
so, with this we change the semantic of the last parameter. it is no longer an output variable but an input variable that should provide an address of the buffer to hold the string, right?
```
0AF4=4,read_string_from_ini_file %1s% section %2s% key %3s% buf_address %4d%
```

Reply:
Also, how do other opcodes work, that also (could) return strings, for instance 0AD3 or 0AD4

Reply:
Yes, same as in SA, except III\VC do not have long/short string types, so only "target pointer" argument is supported.

0AD3 - already threats whatever is provided as output ptr.
0AD4 - perhaps needs fixes, as it can return both numbers and strings. Currently it just writes variable's value.